### PR TITLE
feat: implement inpage navigation - INNO-1555

### PR DESCRIPTION
**Pull request by haringsrob**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] I have put the vanilla component as `devDependencies`
- [x] I have put the specs package as `devDependencies`
- [x] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [x] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [x] My variables naming follow the guidelines (snake case for twig)
- [x] I have provided tests
- [x] I have provided documentation (for the "notes" tab)
- [x] If my local `yarn.lock` contains changes, I have committed it
- [x] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-inpage-navigation/inpage-navigation.story.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import iconPath from '@ecl/ec-resources-icons/dist/sprites/icons.svg';
+
+import inpageNavigationDocs from './docs/inpage-navigation.md';
+
+import demoData from './demo/data';
+import pageFiller from './demo/page-filler';
+
+import inpageNavigation from './inpage-navigation.html.twig';
+
+storiesOf('Components/Inpage navigation', module)
```
Comment (yhuard) [inline src/ec/packages/ec-component-inpage-navigation/inpage-navigation.story.js]:
```suggestion
storiesOf('Components/Navigation/Inpage navigation', module)
```

the component should be listed under "Navigation" ;)

File: src/ec/packages/ec-component-link/link.html.twig [hunk 2]
```diff
@@ -82,8 +82,7 @@
   {% for attr in extra_attributes %}
     {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
   {% endfor %}
-{% endif %} 
-
```
[the remark this answers is not in the document]
Comment (haringsrob) [reply src/ec/packages/ec-component-link/link.html.twig]: IDE auto removes trailing spaces. Checked it out from base.

**Comment by haringsrob** [comment]
I made the changes requested.

**Comment by haringsrob** [comment]
I dont really notice any "lag" it does skip items when you scroll fast.

**Pull request merged**